openeo/testing/results.py

```diff
@@ -230,9 +233,17 @@ def _compare_xarray_dataarray(
         issues.append(f"Shape mismatch: {actual.shape} != {expected.shape}")
     compatible = len(issues) == 0
     try:
-        xarray.testing.assert_allclose(a=actual, b=expected, rtol=rtol, atol=atol)
+        if (pixel_tolerance > _DEFAULT_PIXELTOL) and compatible:
+            _bad_pixels = (actual * 1.0 - expected * 1.0) > atol
+            assert (
+                _bad_pixels.mean().item() * 100 <= pixel_tolerance
+            ), "Percentage number of pixels that are different is above the threshold"
```

Comment:
put the actual percentage and threshold percentage in the error message to make this a helpful message